pkg/typesystem/typesystem_test.go

```diff
@@ -4411,6 +4425,192 @@ func TestHasTypeInfo(t *testing.T) {
 	}
 }
 
+func TestGetAllAuthorizationModelNodes(t *testing.T) {
+	t.Run("error_handling_for_non_AuthorizationModelNode", func(t *testing.T) {
+		dut := multi.NewDirectedGraph()
+		dut.AddNode(simple.Node(1))
+		dut.AddNode(simple.Node(2))
+		_, err := getAllAuthorizationModelNodes(dut)
+		require.Error(t, err)
+	})
+	t.Run("correct_model", func(t *testing.T) {
+		model := `
+model
+	schema 1.1
+type user
+type group
+	relations
+		define member: [user, group#member]
+`
+		transformedModel := testutils.MustTransformDSLToProtoWithID(model)
+		authorizationModelGraph, err := graph.NewAuthorizationModelGraph(transformedModel)
+		require.NoError(t, err)
+		nodes, err := getAllAuthorizationModelNodes(authorizationModelGraph.DirectedGraph)
+		require.NoError(t, err)
+		keys := maps.Keys(nodes)
+		sort.Strings(keys)
+		require.Equal(t, []string{"group", "group#member", "user"}, keys)
+	})
+}
```

Comment:
Like I said in an earlier comment, this doesn't belong here, if you need to look up nodes by ID, then the graph should be kind enough to provide that info to you :) I'll send a PR to `language`

Reply:
Once the change is made to language, I will nuke this :)